### PR TITLE
Wrap RethinkDB objects in a mixin class

### PR DESCRIPTION
**Pull request by McSinyx**
This reduce the fraction when declare new wrapper classes:
It is no longer needed to define (or test) each descriptor individually.

The old test for User is now used as the generic test for the wrapper.

<sup>This PR also serves my thirst for meta programming but let's ignore that.</sup>

### CODE REVIEW

File: tests/test_db_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    async with app.app_context():
+        user, monty = User('silasl'), 'Monty Python'
+        name = await user.name
+        with raises(AttributeError): user.name = monty
+
+        await user.update(name=monty)
+        assert await user.name == monty
+        await user.update(name=name)
+
+        with raises(ValueError): await user.update(kink='DBMS')
```
Comment (Huy-Ngo) [inline tests/test_db_wrapper.py:52]: What will happen when we add other objects (e.g. Project)? Do we write tests for those in these two same functions, or it's not necessary to write any at all?

Comment (McSinyx) [reply tests/test_db_wrapper.py:52]: If this patch is done correctly, no additional test will be needed for subclasses of RethinkObject.  And of course I think it works correctly (hence the PR); it's your job as the reviewer to make sure that is true.

**Review by Huy-Ngo — APPROVED** [review]
It works and I don't see any problem so far.

**Review by 9a24f0 — APPROVED** [review]
LGTM

**Comment by McSinyx** [comment]
Thanks!

**Pull request merged**